projects/packages/autoloader/tests/php/bin/test-coverage.php

```diff
@@ -0,0 +1,247 @@
+	// Find the line that the `class` keyword occurs on so that we can use it to calculate an offset from the header.
+	$content = file_get_contents( $file ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+	// Find the class keyword and capture the number of characters in the string before this point.
+	if ( 0 === preg_match_all( '/^class /m', $content, $matches, PREG_OFFSET_CAPTURE ) ) {
+		return null;
+	}
+
+	// Support both styles of line endings.
+	$newlines = substr_count( $content, "\r\n", 0, $matches[0][0][1] );
+	if ( $newlines > 0 ) {
+		return $newlines + 1;
+	}
+	$newlines = substr_count( $content, "\n", 0, $matches[0][0][1] );
+	if ( $newlines > 0 ) {
+		return $newlines + 1;
+	}
+
+	return null;
+}
+
+/**
+ * Creates a map for converting file paths to src paths.
+ *
+ * @param string[] $report_file_paths An array containing all of the paths for the report.
+ * @return array A map describing how to transform built files into src coverage.
+ */
+function get_path_transformation_map( $report_file_paths ) {
+	// We're going to create a map describing how to transform files to src files.
+	// We're also going to store any metadata needed to perform the merge safetly.
+	$transformation_map = array();
+
+	// Scan the src directory so that we can create the map to convert between files.
+	$raw_src_files = scandir( ROOT_DIR . DIRECTORY_SEPARATOR . 'src' );
+	$src_file_map  = array();
+	foreach ( $raw_src_files as $file ) {
+		// Only PHP files will be copied.
+		if ( substr( $file, -4 ) !== '.php' ) {
+			continue;
+		}
+
+		$file = ROOT_DIR . DIRECTORY_SEPARATOR . 'src' . DIRECTORY_SEPARATOR . $file;
+		if ( ! file_exists( $file ) ) {
+			continue;
+		}
+
+		// We need to use the class keywork to address the line offset from injecting the header.
+		$class_line = count_lines_before_class_keyword( $file );
+		if ( ! isset( $class_line ) ) {
+			// The autoloader only has class files and so this is fine.
+			continue;
+		}
+
+		$src_file_map[ $file ] = array(
+			'file'       => $file,
+			'class_line' => $class_line,
+		);
+	}
+
+	// Create a map describing the file transformations.
+	foreach ( $report_file_paths as $report_file_path ) {
+		// We will use the class line from the report file to calculate the offset from the src file to apply to coverage lines.
+		$class_line = count_lines_before_class_keyword( $report_file_path );
+		if ( ! isset( $class_line ) ) {
+			continue;
+		}
+
+		// Attempt to find the original file.
+		// Note: This does not support nested directories!
+		$src_file_path = null;
+		foreach ( $src_file_map as $src_file ) {
+			// We don't need to perform any transformations if the file path is the same.
+			if ( $src_file['file'] === $report_file_path ) {
+				continue;
+			}
+
+			if ( basename( $src_file['file'] ) === basename( $report_file_path ) ) {
+				$src_file_path = $src_file['file'];
+				break;
+			}
+		}
+		if ( ! isset( $src_file_path ) ) {
```

Comment:
```suggestion
		if ( null === $src_file_path ) {
```
or since no file we care about will be named "0",
```suggestion
		if ( ! $src_file_path ) {
```